linuxPackages_latest.rtl8821ce fails to compile

With kernel 5.10.2, this drivers fails to compile.
```
/build/source/os_dep/osdep_service.c: In function 'isFileReadable':
/build/source/os_dep/osdep_service.c:2203:11: error: implicit declaration of function 'get_fs'; did you mean 'get_sa'? [-Werror=implicit-function-declaration]
 2203 |   oldfs = get_fs();
      |           ^~~~~~
      |           get_sa
/build/source/os_dep/osdep_service.c:2203:11: error: incompatible types when assigning to type 'mm_segment_t' {aka 'struct <anonymous>'} from type 'int'
/build/source/os_dep/osdep_service.c:2204:3: error: implicit declaration of function 'set_fs'; did you mean 'sget_fc'? [-Werror=implicit-function-declaration]
 2204 |   set_fs(KERNEL_DS);
      |   ^~~~~~
      |   sget_fc
/build/source/os_dep/osdep_service.c:2204:10: error: 'KERNEL_DS' undeclared (first use in this function); did you mean 'KERNFS_NS'?
 2204 |   set_fs(KERNEL_DS);
      |          ^~~~~~~~~
      |          KERNFS_NS
/build/source/os_dep/osdep_service.c:2204:10: note: each undeclared identifier is reported only once for each function it appears in
/build/source/os_dep/osdep_service.c: In function 'retriveFromFile':
/build/source/os_dep/osdep_service.c:2241:12: error: incompatible types when assigning to type 'mm_segment_t' {aka 'struct <anonymous>'} from type 'int'
 2241 |    oldfs = get_fs();
      |            ^~~~~~
/build/source/os_dep/osdep_service.c:2242:11: error: 'KERNEL_DS' undeclared (first use in this function); did you mean 'KERNFS_NS'?
 2242 |    set_fs(KERNEL_DS);
      |           ^~~~~~~~~
      |           KERNFS_NS
/build/source/os_dep/osdep_service.c: In function 'storeToFile':
/build/source/os_dep/osdep_service.c:2276:12: error: incompatible types when assigning to type 'mm_segment_t' {aka 'struct <anonymous>'} from type 'int'
 2276 |    oldfs = get_fs();
      |            ^~~~~~
/build/source/os_dep/osdep_service.c:2277:11: error: 'KERNEL_DS' undeclared (first use in this function); did you mean 'KERNFS_NS'?
 2277 |    set_fs(KERNEL_DS);
      |           ^~~~~~~~~
      |           KERNFS_NS
cc1: some warnings being treated as errors
make[3]: *** [/nix/store/vfjkakb00wa967gphgbrw72lhqlhrh6j-linux-5.10.2-dev/lib/modules/5.10.2/source/scripts/Makefile.build:279: /build/source/os_dep/osdep_service.o] Error 1
make[2]: *** [/nix/store/vfjkakb00wa967gphgbrw72lhqlhrh6j-linux-5.10.2-dev/lib/modules/5.10.2/source/Makefile:1805: /build/source] Error 2
make[1]: *** [/nix/store/vfjkakb00wa967gphgbrw72lhqlhrh6j-linux-5.10.2-dev/lib/modules/5.10.2/source/Makefile:185: __sub-make] Error 2
make[1]: Leaving directory '/nix/store/vfjkakb00wa967gphgbrw72lhqlhrh6j-linux-5.10.2-dev/lib/modules/5.10.2/build'
make: *** [Makefile:2244: modules] Error 2
builder for '/nix/store/mh31rcg11x0aarvgvd30rgz69nminqws-rtl8821ce-5.10.2-5.5.2_34066.20200325.drv' failed with exit code 2
```

**To Reproduce**
Steps to reproduce the behavior:
  Compile this driver against the latest Linux kernel
1. Add the following to your `configuration.nix`: `extraModulePackages = with pkgs.linuxPackages_latest; [ rtl8821ce ];`
2. Rebuild the system

**Expected behavior**
The driver compiles, wifi works and te\he system succesfully upgrades

**Notify maintainers**
@hhm0 @samuelgrf 

**Metadata**
 - system: `"x86_64-linux"`
 - host os: `Linux 5.9.12, NixOS, 21.03pre257780.e9158eca70a (Okapi)`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Nix) 2.3.9`
 - channels(root): `"nixos-21.03pre259928.2f47650c2f2"`
 - channels(ivv): `"home-manager, nixpkgs-21.03pre260073.ad13f0e569c"`
 - channels(ivar): `"home-manager, nixpkgs-21.03pre250049.4b30d5601da, nixos-20.09pre231796.22a81aa5fc1"`
 - nixpkgs: `/home/ivv/.nix-defexpr/channels/nixpkgs`

Maintainer information:
```yaml
# a list of nixpkgs attributes affected by the problem
attribute: linuxPackages_latest.rtl8821ce
```

--- a/pkgs/os-specific/linux/rtl8821ce/default.nix
+++ b/pkgs/os-specific/linux/rtl8821ce/default.nix
@@ -1,13 +1,14 @@
 { lib, stdenv, fetchFromGitHub, kernel, bc }:
+
 stdenv.mkDerivation rec {
-  name = "rtl8821ce-${kernel.version}-${version}";
-  version = "5.5.2_34066.20200325";
+  pname = "rtl8821ce-${kernel.version}";
+  version = "unstable-2020-12-16";
 
   src = fetchFromGitHub {
     owner = "tomaspinho";
     repo = "rtl8821ce";
-    rev = "8d7edbe6a78fd79cfab85d599dad9dc34138abd1";
-    sha256 = "1hsf8lqjnkrkvk0gps8yb3lx72mvws6xbgkbdmgdkz7qdxmha8bp";
+    rev = "14b536f0c9ad2d0abbdab8afc7ade684900ca9cf";
+    sha256 = "0z7r7spsgn22gwv9pcmkdjn9ingi8jj7xkxasph8118h46fw8ip2";
   };
 
   hardeningDisable = [ "pic" ];

